doc/contrib.md

```diff
@@ -0,0 +1,157 @@
+# Packetgraph contribution guide
+
+We will try to list here all the things you need to know in order to start
+devlopping on packetgraph and creating new bricks.
+
+First of all, you need to be able to build packetgraph like any user, check
+[packetgraph's main page](https://github.com/outscale/packetgraph).
+
+Here a little tree view of the project:
+```
+.
+|-- doc
+|   `-- contrib      : you are here (X)
+|-- examples
+|   |-- dperf
+|   |-- firewall
+|   |-- rxtx
+|   `-- switch
+|-- include
+|   `-- packetgraph  : public packetgraph API
+|-- src              : all internal source code built in packetgraph lib
+|   |-- npf          : imported code from NPF project
+|   `-- utils
+`-- tests
+    |-- antispoof
+    |-- core
+    |-- diode
+    |-- firewall
+    |-- integration
+    |-- nic
+    |-- pmtud        : Typical brick test folder, containing benchmarks
+    |   |-- bench.c
+    |   |-- bench.sh
+    |   |-- tests.c
+    |   `-- test.sh
+    |-- print
+    |-- queue
+    |-- rxtx
+    |-- style
+    |-- switch
+    |-- tap
+    |-- vhost
+    `-- vtep
+```
+
+# Before opening a pull request
+
+In a pull request, all commits must pass all sections described below
+
+## Build everything
+
+As a packetgraph developper, you need to run your configure command with some
+extra parameters to be sure that your modifications will make example and
+benchmark build:
+```
+./configure --with-examples --with-benchmarks
+make
+```
+
+## Run style check
+
+Be sure that you code pass some style check by running:
+```
+make style
+```
+
+Some style test will run if you have some tools like `rats`, `cppcheck`,
+`flawfinder` or `lizard`. The only blocking test is `checkpatch.pl` which
+is embedded with the project.
+
+## Run tests
+
+To build and run all possible tests:
+```
+make check
+```
+
+You may not run all tests or rebuild everything when developping.
+
+To run a specific test, for example: pmtud brick tests, you just have to call
+the tests/core/test.sh from your build directory.
+```
+./tests/core/test.sh
+```
+
+If you just want to rebuild pmtud tests, run:
+```
+make tests-pmtud
+```
+
+Some tests will need some extra commands to be available on your system like
+`wget`, `qemu-system-x86_64` or `awk`.
+Some tests like vhost will spawn a virtual machine, so you will also need to
+[configure some hugepages on your system](https://github.com/outscale/butterfly#configure-huge-pages).
+
+## Run benchmarks
+
+Like tests, benchmarks must work. You can run all benchmarks by running:
+```
+make bench
+```
+And to run a specific benchmark (still from your build directory):
+```
+./tests/pmtud/bench.sh
+```
+
+# Developping a new brick
+
+## First things to think about
+
+Maybe you will have some question to ask yourself before developping a new
+brick:
+- Is the brick I would like to create only do the job related to it's job ?
```

Comment:
I'm not sure our brick are really kiss(the firewall is everything but kiss), nor peoples will care about this, if someone need a brick he will just create and that's all.